master/internal/proxy/proxy.go

```diff
@@ -106,3 +116,39 @@ func (p *Proxy) newHandler(serviceKey string) echo.HandlerFunc {
 		return nil
 	}
 }
+
+// Service an HTTP CONNECT request, which have a hostname:port in place of a normal route.
+func (p *Proxy) newConnectHandler() echo.HandlerFunc {
+	return func(c echo.Context) error {
+		// Parse the request-target, which must be the bare hostname and port, per RFC 7231.
```

Comment:
> which must be the bare hostname and port

"which must be the bare hostname and port of the target" or something more descriptive. This was one of the only parts that was slightly confusing on the first read.

Reply:
How about:
```
// Parse the request-target, which must be in hostname[:port] format, per RFC 7231.
```
?

Reply:
yep, that's perfect.